tmt/steps/discover/__init__.py

```diff
@@ -1,24 +1,32 @@
 import os
-from typing import Optional, Type
+from typing import TYPE_CHECKING, Any, List, Optional, Type
 
 import click
 from fmf.utils import listed
 
 import tmt
+
+if TYPE_CHECKING:
+    import tmt.steps
+    import tmt.base
+    import tmt.options
+
 import tmt.utils
 
 
 class Discover(tmt.steps.Step):
     """ Gather information about test cases to be executed. """
 
-    def __init__(self, plan, data):
+    data: List[tmt.steps.StepData]
+
+    def __init__(self, plan: tmt.base.Plan, data: tmt.steps.StepData):
```

Comment:
These will be the same case, `tmt.base` is still being imported when `tmt.steps.discover` makes these references. Try forward references instead, i.e. `'tmt.base.Plan'`.

Reply:
it is resolved